src/seedu/addressbook/data/person/UniquePersonList.java

```diff
@@ -231,6 +232,36 @@ public Boolean containsPersonWithUsername(String username) {
         }
     }
 
+    /** Iterates through the UniquePersonList to check the attendance of each person
+     * @param date of which list of present people should generate
+     * @return A list of present people
+     */
+    public List listOfPresentPeople (String date) {
+        List <Name> listOfPresent = new ArrayList<>();
+        for (Person p: internalList) {
+            Boolean isPresent = p.viewAttendanceDateMethod(date);
+            if (isPresent) {
+                listOfPresent.add(p.getName());
+            }
+        }
+        return listOfPresent;
+    }
```

Comment:
Er..... You claim to return a list of people, but you are returning a list of names here. Even the function header is wrongly typed.
Also methods are wrongly named, method should sound like verbs, not nouns